### PR TITLE
Make sure 'size' parameter is only used for input types that support it (product quantity selector)

### PR DESCRIPTION
**Pull request by kurzi2704**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

The attribute `size` is not allowed on input type `number` elements. https://screenshot.sunlime.at/6c130f56a61a652a730124ace6f98329.png 

This PR aims to get rid of that html issue. 

https://www.w3schools.com/tags/att_input_size.asp

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a product in WooCommerce > Products > Add Product.
2. Go to the product page.
3. Using your browser dev tools inspect the HTML for the quantity selector and confirm that it doesn't have a `size` attribute:
   Screenshot 2024-08-29 at 08 49 24
   Screenshot 2024-08-29 at 08 49 19
4. Enable the following snippet on your site by using the [Code Snippets plugin](https://wordpress.org/plugins/code-snippets/) or dropping the following as a .php file inside `wp-content/mu-plugins/`:
   ```php
   <?php
   add_filter( 'woocommerce_quantity_input_type', fn() => 'text' );
   ```
5. Refresh the product page.
6. As in step 3, inspect the quantity selector with the devtools, but this time confirm that the HTML contains a `size="4"` attribute:
   Screenshot 2024-08-29 at 08 49 09

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

fixed html error by having attribute size on input type number. 

</details>

### CODE REVIEW

File: plugins/woocommerce/templates/global/quantity-input.php
Comment (senadir) [inline plugins/woocommerce/templates/global/quantity-input.php]: Version should be that of next release, so 9.3.0
```suggestion
 * @version 9.3.0
```

**Review by senadir — APPROVED** [review]
Looking good, will add a changelog and merge it.

File: plugins/woocommerce/templates/global/quantity-input.php
Comment (kurzi2704) [reply plugins/woocommerce/templates/global/quantity-input.php]: Thank you! :)

**Review by jorgeatorres — APPROVED** [review]
I've pushed a fix for some linting issues but this looks good to me. Waiting for CI before merging.

**Comment by rodelgc** [comment]
Hi @jorgeatorres @senadir @kurzi2704  could you please provide testing instructions for this PR? Thanks!

**Comment by kurzi2704** [comment]
@rodelgc the size attribute is only allowed to be added to text, search, tel, url, email, and password input types, but not on any other input types in that specific case the size attribute must not be added to the input type number for the quantity input ->  https://www.w3schools.com/tags/att_input_size.asp 

I am not sure how you are going to test it (manually or automated), but in the end, the quantity input should not have a size attribute anymore.

**Comment by jorgeatorres** [comment]
I've added some more comprehensive testing instructions. Thanks for the ping @rodelgc!

**Pull request merged**
